src/middlewared/middlewared/plugins/account_/2fa.py

```diff
@@ -13,11 +14,13 @@ async def provisioning_uri(self, username):
         Returns the provisioning URI for the OTP for `username`. This can then be encoded in a QR code and used
         to provision an OTP app like Google Authenticator.
         """
-        user = await self.middleware.call('user.query', [['username', '=', username]], {'get': True})
+        user = await self.translate_username(username)
```

Comment:
This now raises a `CallError` instead of a `MatchNotFoundError`. Is this desired?

Reply:
I think it is fine, this is a new endpoint and not being used before so there is no backwards compatibility to lookout for.